Module/src/NewWordCloudCommand.cs

```diff
@@ -0,0 +1,1001 @@
+
+namespace PSWordCloud
+{
+    /// <summary>
+    /// Defines the New-WordCloud cmdlet.
+    ///
+    /// This command can be used to input large amounts of text, and will generate a word cloud based on
+    /// the relative frequencies of the words in the input text.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "WordCloud", DefaultParameterSetName = "ColorBackground")]
+    [Alias("wordcloud", "nwc", "wcloud")]
+    public class NewWordCloudCommand : PSCmdlet
+    {
+
+        #region Constants
+
+        private const float FOCUS_WORD_SCALE = 1.3f;
+        private const float BLEED_AREA_SCALE = 1.15f;
+        private const float MIN_SATURATION_VALUE = 5f;
+        private const float MIN_BRIGHTNESS_DISTANCE = 25f;
+        private const float MAX_WORD_WIDTH_PERCENT = 0.75f;
+        private const float PADDING_BASE_SCALE = 0.05f;
+
+        internal const float STROKE_BASE_SCALE = 0.02f;
+
+        #endregion Constants
+
+        #region StaticMembers
+
+        private static readonly string[] _stopWords = new[] {
+            "a","about","above","after","again","against","all","am","an","and","any","are","aren't","as","at","be",
+            "because","been","before","being","below","between","both","but","by","can't","cannot","could","couldn't",
+            "did","didn't","do","does","doesn't","doing","don't","down","during","each","few","for","from","further",
+            "had","hadn't","has","hasn't","have","haven't","having","he","he'd","he'll","he's","her","here","here's",
+            "hers","herself","him","himself","his","how","how's","i","i'd","i'll","i'm","i've","if","in","into","is",
+            "isn't","it","it's","its","itself","let's","me","more","most","mustn't","my","myself","no","nor","not","of",
+            "off","on","once","only","or","other","ought","our","ours","ourselves","out","over","own","same","shan't",
+            "she","she'd","she'll","she's","should","shouldn't","so","some","such","than","that","that's","the","their",
+            "theirs","them","themselves","then","there","there's","these","they","they'd","they'll","they're","they've",
+            "this","those","through","to","too","under","until","up","very","was","wasn't","we","we'd","we'll","we're",
+            "we've","were","weren't","what","what's","when","when's","where","where's","which","while","who","who's",
+            "whom","why","why's","with","won't","would","wouldn't","you","you'd","you'll","you're","you've","your",
+            "yours","yourself","yourselves" };
+
+        private static readonly char[] _splitChars = new[] {
+            ' ','\n','\t','\r','.',',',';','\\','/','|',
+            ':','"','?','!','{','}','[',']',':','(',')',
+            '<','>','“','”','*','#','%','^','&','+','=' };
+
+        private static readonly object _randomLock = new object();
+        private static Random _random;
+        private static Random Random
+        {
+            get
+            {
+                lock (_randomLock)
```

Comment:
If you're trying to attain thread safety, you need to either wrap the calls to `Next*()` in a lock _or_ maintain a per-thread instance - wrapping the getter in a lock won't help you :)

Reply:
Whaaat why nooooot :(

Reply:
Imagine two different threads each doing this:

```
// enter lock scope
var staticRandom = this.Random;
// exit lock scope

// no longer locked, yolo
staticRandom.Next()
```

Reply:
_oh_.

Yolo indeed....

Reply:
[Here's a really nice example utilizing thread-local instances](https://stackoverflow.com/a/11109361/712649):

```csharp
public class ThreadSafeRandom
{
    private static readonly Random _global = new Random();
    [ThreadStatic] private static Random _local;

    public ThreadSafeRandom()
    {
        if (_local == null)
        {
            lock (_global)
            {
                if (_local == null)
                {
                    int seed = _global.Next();
                    _local = new Random(seed);
                }
            }
        }
    }

    public int Next()
    {
        return _local.Next();
    }
}
```

Now you just need to add your `NextFloat()` wrapper (utilize the lock in the same manner as in the `Next()` wrapper), and you've gotten yourself a thread-safe `Random` with minimal overhead :)

Reply:
🤔 This will need some thought. I want users to be able to specify the seed, and I can't really do that if it's threadstatic.

Maybe I can? 🤔

Reply:
Oh, I see a way. I think.

Reply:
Hmm. I'll keep this in mind for later. I don't think this is worth the effort at the moment... :/